src/mixins/itext_behavior.mixin.js

```diff
@@ -28,6 +28,8 @@
       this.on('added', function() {
         var canvas = _this.canvas;
         if (canvas) {
+          _this._restoreClickHandlerProps();
```

Comment:
```suggestion
          this._clickHandlerInitialized = false;
```

I would love to have just this and a small test that can verify it works. Do we need to check if the canvas is the exact same canvas?

Reply:
I also wonder if today we do multiple registarations.

Reply:
@asturur Sorry for taking so long to get back on this. I've pushed the suggested change with an update to the test. To answer about checking if the canvas is exactly the same, I don't think this is necessary as I believe the listener would need to be added regardless of if the canvas was the same as previous